x-pack/plugin/esql/src/main/antlr/EsqlBaseLexer.g4

```diff
@@ -34,13 +34,15 @@ WS
     : [ \r\n\t]+ -> channel(HIDDEN)
     ;
 
-fragment INDEX_UNQUOTED_IDENTIFIER_PART
-    : ~[=`|,[\]/ \t\r\n]
+// in 8.14 ` were not allowed
+// this has been relaxed in 8.15 since " is used for quoting
+fragment UNQUOTED_SOURCE_PART
+    : ~["=|,[\]/ \t\r\n]
     | '/' ~[*/] // allow single / but not followed by another / or * which would start a comment
```

Comment:
I noticed that this is maybe a bit too permissive. For instance, adding the following to the statement parser tests passes just fine:
```
assertStringAsIndexPattern("foo/=", "from foo/=");
```
This example also works for the other symbols that are forbidden in line 40: prepending `/` allows all of them except `/`, in particular even `"`.

Reply:
These patterns aren't legal as index names, but we do allow them, indeed. Not sure if this is an issue in itself, but we could follow up on this (this hasn't been changed in this PR).